extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/export/PrometheusRegistryProcessor.java

```diff
@@ -38,13 +39,17 @@ public boolean getAsBoolean() {
     }
 
     @BuildStep(onlyIf = PrometheusEnabled.class)
-    MicrometerRegistryProviderBuildItem createPrometheusRegistry(
+    MicrometerRegistryProviderBuildItem createPrometheusRegistry(MicrometerConfig config,
             BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
 
-        // Add the Prometheus Registry Producer
-        additionalBeans.produce(AdditionalBeanBuildItem.builder()
+        // Add the Prometheus Registry beans
+        AdditionalBeanBuildItem.Builder builder = AdditionalBeanBuildItem.builder()
                 .addBeanClass(PrometheusMeterRegistryProvider.class)
-                .setUnremovable().build());
+                .setUnremovable();
+        if (config.export.prometheus.defaultRegistry) {
+            builder.addBeanClass(PrometheusMeterRegistryProducer.class);
```

Comment:
This may not be exactly right (memory fails). Could also use a constant here. But the Producer is free to reference the Micrometer PrometheusRegistry, which may not always be present.

```suggestion
            builder.addBeanClass("io.quarkus.micrometer.runtime.export.PrometheusMeterRegistryProducer");
```